### PR TITLE
RACE: minor changes to demo-recording

### PR DESCRIPTION
**Pull request by meag**
k_race_autorecord set by default
If not recording demo, don't set demoname in .top file
"/race_dl_record_demo x" downloads demo of record #x
Fixed up route names for dmX maps

### CODE REVIEW

File: src/race.c [hunk 1]
```diff
@@ -2060,6 +2063,38 @@ void r_timeout( )
...
+	trap_CmdArgv( 1, arg_1, sizeof( arg_1 ) );
+
+	record = atoi( arg_1 );
+
+	if ( record )
+		record = 1;
+
+	record = bound(0, atoi( arg_1 ) - 1, NUM_BESTSCORES - 1 );
+
+	if ( race.records[record].time > 999998 )
```
Comment (qqshka) [inline src/race.c]: I guess its time to replace 999998 with #define.

File: src/race.c [hunk 2]
```diff
@@ -2060,6 +2063,38 @@ void r_timeout( )
...
+		return;
+
+	trap_CmdArgv( 1, arg_1, sizeof( arg_1 ) );
+
+	record = atoi( arg_1 );
+
+	if ( record )
+		record = 1;
+
+	record = bound(0, atoi( arg_1 ) - 1, NUM_BESTSCORES - 1 );
```
Comment (qqshka) [inline src/race.c]: record = bound(0, record, NUM_BESTSCORES - 1 );

File: src/race.c [hunk 3]
```diff
@@ -2060,6 +2063,38 @@ void r_timeout( )
...
+	int		record;
+
+	if ( !race_command_checks() )
+		return;
+
+	trap_CmdArgv( 1, arg_1, sizeof( arg_1 ) );
+
+	record = atoi( arg_1 );
+
+	if ( record )
```
Comment (qqshka) [inline src/race.c]: Do you mean if ( ! record ) here instead?

File: src/race.c (hunk 2 above)
Comment (qqshka) [inline src/race.c]: trap_CmdArgv( 1, arg_1, sizeof( arg_1 ) );
record = bound(0, atoi( arg_1 ) - 1, NUM_BESTSCORES - 1 );

**Comment by meag** [comment]
Thanks @qqshka, have tidied up a bit and updated.

**Pull request merged**